Automattic/jetpack
# JS Components: Add JetpackSearchLogo component

Fixes #26451 

#### Changes proposed in this Pull Request:

* Introduce the `JetpackSearchLogo` component
* Replace the header of Jetpack Search upsell page with `JetpackSearchLogo`
* Replace the footer link of Jetpack Search upsell page with [Jetpack Search product page link](https://jetpack.com/upgrade/search/)

> Note: This JetpackSearchLogo component could probably be extended to a Jetpack plugin logo component in the future, thus I introduce the `pluginName` prop to highlight this intention.

#### Other information:


#### Jetpack product discussion


p1HpG7-icd-p2

#### Does this pull request change what data or activity we track or use?


N/A

#### Testing instructions:






Storybook:

* Run `cd projects/js-packages/storybook && npm run storybook:dev` and confirm that you can see `JetpackSearchLogo` component in Storybook.
* Ensure that the `JetpackSearchLogo` component works as described.

Upsell Page:
* Apply this change on a Jetpack site without a Search subscription.
* Navigate to `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`.
* Ensure the header section display with the new Jetpack Search logo.

| Before | After |
| --- | ----------- |
| <img width="1266" alt="old" src="https://user-images.githubusercontent.com/6869813/192574759-76139f13-d32e-45f2-9a59-7b747445fdd8.png"> | <img width="1335" alt="new" src="https://user-images.githubusercontent.com/6869813/192574807-41f45982-3543-4abb-9378-43305ad6aba5.png"> |
* Ensure the footer link for the Jetpack Search logo is navigating to `https://jetpack.com/upgrade/search/`.
